backend/src/apiserver/resource/resource_manager.go

```diff
@@ -211,6 +211,15 @@ func (r *ResourceManager) CreateRun(apiRun *api.Run) (*model.RunDetail, error) {
 	// Append provided parameter
 	workflow.OverrideParameters(parameters)
 
+	// Mark auto-added artifacts as optional. Otherwise workflows fail in Argo 2.3. Proper fix: Fix the components to explicitly declare the artifacts where needed. Fix the compiler to stop auto-adding atrifacts to all tasks.
```

Comment:
// Mark auto-added artifacts as optional. Otherwise workflows fail in Argo 2.3. 
// TODO: Fix the components to explicitly declare the artifacts where needed. Fix the compiler to stop auto-adding atrifacts to all tasks.

Reply:
Done.